src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs

```diff
@@ -74,55 +77,46 @@ public class CollateralChecker : ICollateralChecker
             this.depositsByAddress = new Dictionary<string, Money>();
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.blockStoreClient = new BlockStoreClient(loggerFactory, httpClientFactory, $"http://{settings.CounterChainApiHost}", settings.CounterChainApiPort);
-            this.isInitialized = false;
         }
 
         public async Task InitializeAsync()
         {
             this.memberAddedToken = this.signals.Subscribe<FedMemberAdded>(this.OnFedMemberAdded);
             this.memberKickedToken = this.signals.Subscribe<FedMemberKicked>(this.OnFedMemberKicked);
 
-            foreach (CollateralFederationMember federationMember in this.federationManager.GetFederationMembers()
-                .Cast<CollateralFederationMember>().Where(x => x.CollateralAmount != null && x.CollateralAmount > 0))
+            foreach (CollateralFederationMember federationMember in this.federationManager.GetFederationMembers().Cast<CollateralFederationMember>().Where(x => x.CollateralAmount != null && x.CollateralAmount > 0))
             {
                 this.logger.LogDebug("Initializing federation member {0} with amount {1}.", federationMember.CollateralMainchainAddress, federationMember.CollateralAmount);
-
-                this.depositsByAddress.Add(federationMember.CollateralMainchainAddress, 0);
+                this.depositsByAddress.Add(federationMember.CollateralMainchainAddress, federationMember.CollateralAmount);
             }
 
-            while (true)
+            while (!this.cancellationSource.IsCancellationRequested && !this.collateralUpdated)
             {
-                bool success = await this.UpdateCollateralInfoAsync(this.cancellationSource.Token).ConfigureAwait(false);
+                this.collateralUpdated = await this.UpdateCollateralInfoAsync(this.cancellationSource.Token).ConfigureAwait(false);
+                if (this.collateralUpdated)
+                    break;
 
-                if (!success)
+                this.logger.LogWarning(CollateralCheckFailedMessage);
```

Comment:
this can go in `UpdateCollateralInfoAsync` and `CollateralCheckFailedMessage` constant can be removed

Reply:
👍